### ISSUE TITLE
Is the game godot 4.0 compatable?

### ISSUE THREAD

**Issue by Aayan-ta24** [dulvui/ball2box#6]
I tried to setup the game on the latest godot engine 4.0. It gave me a few errors, it might be because of an older engine. 

at first there are some dependencies issue with a. theme.tres, b. SplashScreen.gd

and additional errors like 
BallMachine.gd
```
W 0:00:00:0025   DisplayServerX11: Error setting locale modifiers
  <C++ Source>   platform/linuxbsd/x11/display_server_x11.cpp:5355 @ DisplayServerX11()

W 0:00:00:0025   DisplayServerX11: XOpenIM failed
  <C++ Source>   platform/linuxbsd/x11/display_server_x11.cpp:5401 @ DisplayServerX11()

W 0:00:00:0041   _create_window: XCreateIC couldn't create wd.xic
  <C++ Source>   platform/linuxbsd/x11/display_server_x11.cpp:5108 @ _create_window()
```
what could be the issue?

**Comment by dulvui** [comment]
Yes you are right, currently it is only compatible with Godot 3.x versions and I'm using 3.5.2
Godot 4 has breaking changes, so many things won't work and need to be rewritten/fixed.
Once Godot 4 is quite stable, I will migrate it. Probably somewhere around version 4.1.0 or 4.2.0
But if you want to try to fix it already, pull requests are always welcome.

**Comment by dulvui** [comment]
Migration to Godot 4.1.2 started now

**Comment by dulvui** [comment]
I retried importing the game into Godot 4.2.2, and after some fixes and migrations, it runs now.

This is a screenshots of the game running, and as you can see, it is quite broken.
![image](https://github.com/user-attachments/assets/dcb9a9ae-a8a4-48be-88b0-6c15f58ef5ef)

I'll spend some more time to see how much effort it is to make it work.
Another option could be "Ball2Box 2" or "Ball3Box", created from scratch in Godot 4.2.

**Comment by xolatgames** [comment]
Hi! ✌🏼 I want to contribute your projects, and already have create a branch for it _(ilyuha-godot4)_ :

![Image](https://github.com/user-attachments/assets/a716bd64-c899-4da7-a9db-ed59393c3807)

But I can't publish it, because I haven't permission for it:

![Image](https://github.com/user-attachments/assets/ae272776-7550-4f9b-98e6-7de2a231feeb)

```
ERROR: Permission to dulvui/ball2box.git denied to xolatgames.
fatal: Не удалось прочитать из внешнего репозитория.

Удостоверьтесь, что у вас есть необходимые права доступа
и репозиторий существует.
```

Or in English:

```
ERROR: Permission to dulvui/ball2box.git denied to xolatgames.
fatal: Could not read from external repository.

Make sure you have the necessary access rights
and the repository exists.
```

What can I do to contribute your project? 🙃

**Comment by xolatgames** [comment]
https://github.com/user-attachments/assets/07c6bc7e-4744-4c3e-bce5-0afde55d6a87

**Comment by xolatgames** [comment]
https://github.com/user-attachments/assets/5053ac40-cb67-4ad8-914b-f206b50ee33d

**Comment by xolatgames** [comment]
And also returned some materials in some places:

![Image](https://github.com/user-attachments/assets/801d2a06-21bf-4566-9159-2659e6390bc7)

![Image](https://github.com/user-attachments/assets/845396c8-b89f-4c53-8b87-b7568716786f)

**Comment by xolatgames** [comment]
https://github.com/user-attachments/assets/ae8533a6-45d7-4c43-94b4-ac4bc460ec65

It was fun! 😄 So, can I contribute? 🙃

**Comment by dulvui** [comment]
Cool great work!

Yes you can contribute, but for that you need to create a fork and there you have the permission to do everything.
Than you can create a pull request from your fork to my repository and if everything looks fine, I will merge it.

Here the official Github guide on how to contribute to a project https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project

If you have questions, I'm happy to help

**Comment by xolatgames** [comment]
> [@xolatgames](https://github.com/xolatgames) Cool great work!
> 
> Yes you can contribute, but for that you need to create a fork and there you have the permission to do everything. Than you can create a pull request from your fork to my repository and if everything looks fine, I will merge it.
> 
> Here the official Github guide on how to contribute to a project https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project
> 
> If you have questions, I'm happy to help

Thanks so much! I'll do it now 😉

**Comment by dulvui** [comment]
@xolatgames I tried to merge the `main` branch, that had a lot of changes in the meantime onto `godot4` branch and there are many many problems. I fixed many other bugs in the meantime that miss in the godot4 branch and merging everything is quite a mess.

I'm really sorry, but I think the best way would be to restart the migration to Godot 4 from the main branch, so your work would get lost, but at least I could replicate the steps you did to get it running in Godot 4.
I think I can just look at your commits and see what changed, except you want to try again from the main branch.

**Comment by xolatgames** [comment]
> [@xolatgames](https://github.com/xolatgames) I tried to merge the `main` branch, that had a lot of changes in the meantime onto `godot4` branch and there are many many problems. I fixed many other bugs in the meantime that miss in the godot4 branch and merging everything is quite a mess.
> 
> I'm really sorry, but I think the best way would be to restart the migration to Godot 4 from the main branch, so your work would get lost, but at least I could replicate the steps you did to get it running in Godot 4. I think I can just look at your commits and see what changed, except you want to try again from the main branch.

As you want 🙂

**Comment by dulvui** [comment]
@xolatgames I will try this week to upgrade the current main branch to Godot 4.4 and see what happens.

**Comment by dulvui** [comment]
This repo is getting archived here on Github and moved to Codeberg https://codeberg.org/dulvui/ball2box/
The issue got also moved to Codeberg https://codeberg.org/dulvui/ball2box/issues/3

Please open any new issues there, thank you!
